internal/dao/tests/pubkey_resource_test.go

```diff
@@ -74,6 +75,13 @@ func TestGetResourceByProviderType(t *testing.T) {
 	assert.Equal(t, resource, createdResource)
 }
 
+func TestGetPubkeyResourceByProviderTypeNoRows(t *testing.T) {
```

Comment:
I'd rather see more test in one function by `t.Run()` if those are for the same function, I've tried to start this, but we do not have it everywhere, so up to you, but I'd love to have test function per function mapping 1:1 if possible.

Reply:
Sure.